chroma_core/models/host.py

```diff
@@ -1647,17 +1649,25 @@ def get_deps(self):
         return DependAll(
             [DependOn(host.lnet_configuration, "lnet_up") for host in target_primary_hosts]
             + [DependOn(fs, "stopped") for fs in filesystems]
-            + [DependOn(t, "unmounted") for t in targets]
+            + [DependOn(t, "unmounted") for t in targets if not issubclass(t.downcast_class, ManagedMgs)]
+            + [DependOn(t, "mounted") for t in targets if issubclass(t.downcast_class, ManagedMgs)]
         )
 
     def create_locks(self):
+        from chroma_core.models.target import ManagedMgs
+
         locks = []
         filesystems, targets = self._targets_on_hosts()
 
         for target in targets:
-            locks.append(
-                StateLock(job=self, locked_item=target, begin_state="unmounted", end_state="unmounted", write=True)
-            )
+            if not issubclass(target.downcast_class, ManagedMgs):
```

Comment:
nit:  could remove `not` and swap branches.

Reply:
Done